formulaire/src/main/java/fr/openent/formulaire/export/FormQuestionsExportPDF.java

```diff
@@ -62,298 +67,308 @@ public FormQuestionsExportPDF(HttpServerRequest request, Vertx vertx, JsonObject
+                .filter(imageInfo -> imageInfo.containsKey(ID) && imageInfo.containsKey(DATA))
+                .forEach(imageInfo -> localChoicesMap.put(imageInfo.getString(ID), imageInfo.getString(DATA)));
+
+        // Affect images to corresponding choices (we have to iterate like previously to keep the same order)
+        choicesInfos.getJsonArray(QUESTIONS_CHOICES).stream()
+                .filter(Objects::nonNull)
+                .map(JsonObject.class::cast)
+                .filter(choice -> choice.containsKey(IMAGE))
+                .forEach(choice -> {
+                    String choiceImageId = getImageId(choice);
+                    String imageData = localChoicesMap.get(choiceImageId);
+                    choice.put(PARAM_CHOICE_IMAGE, imageData);
+                });
+    }
+
+
+    /**
+     * Fill questions with their choices and sections with their questions
+     * @param questionChoices all the datas from the choices
+     * @param mapSections a map with id of section as key and sectionsInfos as value
+     * @param mapQuestions a map with id of question as key and questionsInfos as value
+     * @param imageInfos all the datas from the images
+     */
+    private void fillChoices(JsonObject questionChoices, Map<Integer, JsonObject>mapSections, Map<Integer, JsonObject>mapQuestions, List<Future> imageInfos){
+        questionChoices.getJsonArray(QUESTIONS_CHOICES).stream()
+                .filter(Objects::nonNull)
+                .map(JsonObject.class::cast)
+                .forEach(choice -> {
+                    if(choice.containsKey(IMAGE))imageInfos.add(getImageData(choice));
+                    Integer nextQuestionId = choice.getInteger(NEXT_FORM_ELEMENT_ID);
+                    if (nextQuestionId != null) {
+                        String titleNext = null;
+                        JsonObject nextQuestion = mapQuestions.get(nextQuestionId);
+                        JsonObject nextSection = mapSections.get(nextQuestionId);
+                        if (nextQuestion != null) {
+                            titleNext = nextQuestion.getString(TITLE);
+                        } else if (nextSection != null) {
+                            titleNext = nextSection.getString(TITLE);
+                        }
+                        if (titleNext != null) {
+                            choice.put(TITLE_NEXT, titleNext);
+                        }
+                    } else {
+                        choice.put(TITLE_NEXT, I18nHelper.getI18nValue(I18nKeys.END_FORM, request));
+                    }
+                });
+    }
+
+
+    /**
+     * Handle Matrix questions and its children
+     * @param questionsInfos all questionsInfos when not filled yet
+     * @param matrixChildren columns and lines of a matrix question
+     */
+    private void fillMatrixQuestions(JsonArray questionsInfos, JsonArray matrixChildren){
+        List<JsonObject> childrenList = matrixChildren.getList();
```

Comment:
Unchecked type quand tu get `getList`

Le risque c'est que je peux t'envoyer un JsonArray contenu que des list de string

Tu peux "résoudre" ce problème en stream : 

```suggestion
         List<JsonObject> childrenList = matrixChildren.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .collect(Collectors.toList());
```

Tu peux même rajouter ton sort dans ton stream à la suite : 

```
   List<JsonObject> childrenList = matrixChildren.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .sorted(Comparator.nullsFirst(Comparator.comparingInt(a -> a.getInteger(MATRIX_POSITION))))
                .collect(Collectors.toList());
```